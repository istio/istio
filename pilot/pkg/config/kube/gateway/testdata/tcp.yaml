apiVersion: gateway.networking.k8s.io/v1beta1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    port: 34000
    protocol: TCP
    allowedRoutes:
      namespaces:
        from: All
---
# Test we can bind to two parents
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: gateway2
  namespace: istio-system
spec:
  addresses:
  - value: istio-ingressgateway
    type: Hostname
  gatewayClassName: istio
  listeners:
  - name: default
    port: 34000
    protocol: TCP
    allowedRoutes:
      namespaces:
        from: All
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: TCPRoute
metadata:
  name: tcp
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: istio-system
  - name: gateway2
    namespace: istio-system
  rules:
  - backendRefs:
    - name: httpbin
      port: 9090
