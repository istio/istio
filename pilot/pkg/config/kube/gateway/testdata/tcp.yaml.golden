apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*'
    port:
      name: default
      number: 34000
      protocol: TCP
---
apiVersion: networking.istio.io/v1
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway2/default.istio-system
  creationTimestamp: null
  name: gateway2-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*'
    port:
      name: default
      number: 34000
      protocol: TCP
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: TCPRoute/tcp.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: tcp-tcp-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - '*'
  tcp:
  - route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 9090
---
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: TCPRoute/tcp.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: tcp-tcp-1-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway2-istio-autogenerated-k8s-gateway-default
  hosts:
  - '*'
  tcp:
  - route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 9090
---
